src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,115 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const filter = useAppSelector(state => state.filter);
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const filteredTodos: Todo[] | undefined = useMemo(() => {
+    const newArray = todos?.filter(todo => {
+      const lowerCase = todo.title.toLowerCase();
+      const filterLowerCase = filter.query?.toLowerCase() || '';
+
+      switch (filter.status) {
+        case 'active':
+          return !todo.completed && lowerCase.includes(filterLowerCase);
+        case 'completed':
+          return todo.completed && lowerCase.includes(filterLowerCase);
```

Comment:
we can move this condition to some variable
` lowerCase.includes(filterLowerCase)`